app/helpers/date_validation_helper.rb

```diff
@@ -1,7 +1,10 @@
 module DateValidationHelper
   def valid_or_invalid_date(year, month)
     date_args = [year, month, 1].map(&:to_i)
-    if year.present? && Date.valid_date?(*date_args)
```

Comment:
I wonder for simplicity if we could do something similar to:
```
  def valid_or_invalid_date(year, month)
    date_args = [year, month, 1].map(&:to_i)
    Date.new(*date_args)
  rescue ArgumentError, RangeError
    Struct.new(:day, :month, :year).new(1, month, year)
  end
```
we might need to raise an error if the year is not present as well. Im keen on avoiding any explicit checks on the month as we might miss edge cases, let me know your thoughts.